src/components/rank.js

```diff
@@ -38,37 +38,66 @@ export class Rank extends React.Component {
   
   loadChars = async (charsToLoad) => {
     let loadedCharList = [];
-    await asyncForEach(charsToLoad, async char => {
+    await asyncForEach(charsToLoad, async (char) => {
       let charInfo = await this.getChar(char);
-      loadedCharList = loadedCharList.concat(charInfo);
+      if (charInfo) {
+        loadedCharList.push(charInfo); // Solo se agregan objetos válidos
+      }
     });
-    loadedCharList.sort((a, b) => b.level - a.level);
-    loadedCharList = this.rankChars(loadedCharList);
+  
+    if (loadedCharList.length === 0) {
+      console.error("No se pudieron cargar personajes válidos.");
+      return [];
+    }
+  
+    loadedCharList.sort((a, b) => b.level - a.level); // Ordena por nivel
+    loadedCharList = this.rankChars(loadedCharList); // Asigna rangos
     return loadedCharList;
-  }
-
+  };
   // getChar() returns the object with the char's info
 
   getChar = async (charName) => {
-    let targetUrl = `https://api.tibiadata.com/v2/characters/${charName}.json`;
-    let blob = await fetch(targetUrl);
-    let data = await blob.json();
-    return data["characters"]["data"];
-  }
+    let targetUrl = `https://api.tibiadata.com/v4/character/${charName}`;
+    try {
+      let response = await fetch(targetUrl);
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      let data = await response.json();
+  
+      if (data && data.character && data.character.character) {

```

Comment:
Esto es lo mismo que

```suggestion
      if (data?.character?.character) {
```

Si es que alguno de los que tienen el ? no existe, devuelve undefined y no tira error, no se cumple la condición